src/api/pokemons.js

```diff
@@ -0,0 +1,40 @@
+import fetch from "cross-fetch"
+
+const POKEMON_API_BASE_URL = process.env.POKEMON_API_BASE_URL || "https://pokeapi.co/api/v2";
+
+export async function getPokemonById(id) {
+    const getTypeIdFromUrl = (url) => {
+        const regex = /\/type\/(\d+)\/$/;
+        return Number(regex.exec(url)[1])
+    };
```

Comment:
I wasn't aware of this complexity in code, but I am glad you managed to overcome it, cool!

However you should know that extracting this typeId from url sounds like a bad code example for me. We shouldn't do things like that, but I am glad it works and it even saves two requests to pokeapi, which is a good thing :)

But! There is no need to change that.